commons/src/main/scala/dagr/commons/reflect/ReflectionUtil.scala

```diff
@@ -303,6 +303,14 @@ object ReflectionUtil {
     * the creation of and packaging into collections as necessary.
     */
   private[reflect] def typedValueFromString(resultType: Class[_], unitType: Class[_], value: String*): Try[Any] = Try {
+    // Regardless if the result is a collection, just try building it since it may have a String constructor!
+    if (value.size == 1) {
+      buildUnitFromString(resultType, value.head) match {
+        case Failure(_) => Unit
+        case success    => return success
+      }
+    }
+
```

Comment:
I checked out your branch locally and tried something less ugly.  Instead of this whole block, you can just change the conditional on line 306/314 from:
`if (ReflectionUtil.isCollectionClass(resultType))`
to:
`if (unitType != resultType && ReflectionUtil.isCollectionClass(resultType))`

All the tests, including your new test, pass.